docs/troubleshooting/Performance.mdx

```diff
@@ -79,7 +79,120 @@ module.exports = {
 
 See [Glob pattern in parser's option "project" slows down linting](https://github.com/typescript-eslint/typescript-eslint/issues/2611) for more details.
 
-## The `indent` / `@typescript-eslint/indent` rules
+## Changes to `extraFileExtensions` with `projectService` in your ESLint options
+
+When [`extraFileExtensions`](../packages/Parser.mdx#extrafileextensions) changes while using
+[`projectService`](../packages/Parser.mdx#projectservice), the TypeScript server must perform a full project reload.
+Project reloads can be observed using the [debug environment variable](../packages/typescript-estree/#debugging): `DEBUG='typescript-eslint:typescript-estree:*'`.
+
+```
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[]: after=[ '.vue' ]
+typescript-estree:tsserver:info reload projects.
+typescript-estree:useProgramFromProjectService Extra file extensions updated: [ '.vue' ]
+...
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[ '.vue' ]: after=[]
+typescript-estree:tsserver:info reload projects.
+typescript-estree:useProgramFromProjectService Extra file extensions updated: []
+...
+typescript-estree:useProgramFromProjectService Updating extra file extensions: before=[]: after=[ '.vue' ]
+typescript-estree:tsserver:info reload projects.
+typescript-estree:useProgramFromProjectService Extra file extensions updated: [ '.vue' ]
+```
+
+As files are received, if `extraFileExtensions` differs from the previous file,
+the TypeScript server is updated to match your ESLint options.
+The TypeScript server incrementally updates the project structure by creating an execution pipeline.
+If the ordering of files received from ESLint happens to frequently alternate the value of `extraFileExtensions`,
+a project reload may occur before the last pipeline completes.
+In turn, this can lead to cascading performance issues.
+
+```
+typescript-estree:tsserver:info Scheduled: /path/to/tsconfig.src.json, Cancelled earlier one +0ms
+typescript-estree:tsserver:info Scheduled: *ensureProjectForOpenFiles*, Cancelled earlier one +0ms
+```
+
+The impact of setting `extraFileExtensions` per file type in a project may be negligible depending on the order files are received.
+That said, complex ESLint options, cached linting results altering which files are evaluated,
```

Comment:
...and trimming out further details that enforce the idea of _"don't let this change"_:

```suggestion
```